framework/py/flwr/server/superlink/fleet/grpc_rere/fleet_servicer.py

```diff
@@ -199,4 +229,26 @@ def PullObject(
             request.object_id,
         )
 
+        state = self.state_factory.state()
+
+        # Abort if the run is not running
+        abort_msg = check_abort(
+            request.run_id,
+            [Status.PENDING, Status.STARTING, Status.FINISHED],
+            state,
+        )
+        if abort_msg:
+            abort_grpc_context(abort_msg, context)
+
+        if request.node.node_id not in state.get_nodes(run_id=request.run_id):
+            # Cancel insertion in ObjectStore
+            context.abort(grpc.StatusCode.FAILED_PRECONDITION, "Unexpected node ID.")
+
+        # Init store
+        store = self.objectstore_factory.store()
+
+        # Fetch from store
+        if content := store.get(request.object_id):
+            return PullObjectResponse(object_content=content)
+
         return PullObjectResponse()
```

Comment:
Same question below: it might be better to `context.abort` instead of returning an empty response.